src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -559,15 +559,7 @@ test_insert (void)
    }
 
    r = mongoc_collection_insert_one (
-      collection, tmp_bson ("{'$hello': 1}"), NULL, NULL, &error);
-   ASSERT (!r);
-   ASSERT_ERROR_CONTAINS (error,
-                          MONGOC_ERROR_COMMAND,
-                          MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "invalid document");
-
-   r = mongoc_collection_insert_one (
-      collection, tmp_bson ("{'a.b': 1}"), NULL, NULL, &error);
+      collection, tmp_bson ("{'': 1}"), NULL, NULL, &error);
```

Comment:
Empty keys are one of the few client-side validation errors we can still rely upon. Invalid UTF-8 characters would also work in theory, but `tmp_bson` refuses to parse JSON with such characters.